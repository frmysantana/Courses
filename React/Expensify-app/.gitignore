# These files will be made when the package.json is installed and  # webpack is run, so they don't need to be hosted.
node_modules/
public/dist/
.env.test
.env.development
.vscode/