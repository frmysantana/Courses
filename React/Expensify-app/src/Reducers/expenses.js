// Expenses reducer
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      // concat doesn't change state itself; it just returns the result of 
      // concatenating state with action.expense
      return [
        ...state,
        action.expense
      ];
    case 'REMOVE_EXPENSE':  
      // filter also doesn't change the state itself.
      return state.filter((expense) => {
        return expense.id !== action.id;
      });
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
          ...expense,
          ...action.updates
          }
        } else {
          return expense
        }
      });
    default:
      return state;
  }
};

export default expensesReducer;