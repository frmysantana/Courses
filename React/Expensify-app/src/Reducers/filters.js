// Filters reducer
const filtersReducerDefaultState = {
  text: '',
  sortBy: 'Date',
  startDate: undefined,
  endDate: undefined
};
  
export default (state = filtersReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_TEXT_FILTER':
      return {
        ...state,
        text: action.text
      };
    case 'SORT_BY_AMOUNT':
      return {
        ...state,
        sortBy: 'Amount'
      };
    case 'SORT_BY_DATE':
      return {
        ...state,
        sortBy: 'Date'
      }
    case 'SET_START_DATE':
      return {
        ...state,
        startDate: action.start
      }
    case 'SET_END_DATE':
      return {
        ...state,
        endDate: action.end
      }
    default:
      return state;
    }
};