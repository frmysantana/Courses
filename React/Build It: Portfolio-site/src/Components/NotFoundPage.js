import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <h1>Page not Found!</h1>
        404 - <Link to="/">Go home!</Link>
    </div>
);

export default NotFoundPage;