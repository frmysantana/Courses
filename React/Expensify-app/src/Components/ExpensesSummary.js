import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../Selectors/expenses';
import selectExpensesTotal from '../Selectors/expenses-total';

export const ExpensesSummary = ({ expensesLength, expensesTotal }) => {
  return (
    <p>Viewing {
      expensesLength
      } {expensesLength === 1 ? 'expense' : 'expenses'} totalling {
      numeral(expensesTotal/100).format('$0,0.00')
      }.
    </p>
  );
};
 
const mapStateToProps = (state) => {
  const filteredExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expensesLength: filteredExpenses.length,
    expensesTotal: selectExpensesTotal(filteredExpenses)
  }
};

export default connect(mapStateToProps)(ExpensesSummary);