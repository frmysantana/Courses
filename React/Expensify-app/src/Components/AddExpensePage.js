import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../Actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    //Line below abstracts: props.dispatch(addExpense(expense));
    this.props.onSubmit(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm 
          onSubmit={this.onSubmit}
        />
      </div>
    )
  };
}

// Used to abstract the above comment's dispatch call
const mapDispatchToProps = (dispatch) => {
  return {
    onSubmit: (expense) => dispatch(addExpense(expense))
  };
};

export default connect(undefined, )(AddExpensePage);