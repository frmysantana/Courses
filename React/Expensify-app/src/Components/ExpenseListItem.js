import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../Actions/expenses';

const ExpenseListItem = (props) => (
  <div>
    <h3>{props.description}</h3>
    <p>{props.amount} - {props.createdAt}</p>
    {console.log(props.id)}
    <button 
      onClick={(e)=>{props.dispatch(removeExpense({id: props.id}));}}
    >
      Remove
    </button>
  </div>
);

export default connect()(ExpenseListItem);