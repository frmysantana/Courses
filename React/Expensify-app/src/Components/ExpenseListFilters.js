import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../Actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilters extends React.Component {
  state = {
    calenderFocused: null
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChange = (calenderFocused) => {
    this.setState(()=>({ calenderFocused }));
  };

  render() {
    return (
      <div>
        <input 
          type="text" 
        value={this.props.filters.text} 
        onChange={(e)=> {
          this.props.dispatch(setTextFilter(e.target.value));
        }}
        />
        <select 
          value={this.props.filters.sortBy} 
          onChange={(e) => {
            if (e.target.value === 'Date') {
              this.props.dispatch(sortByDate());
            } else if (e.target.value === 'Amount') {
              this.props.dispatch(sortByAmount());   
            }
          }}
        >
          <option value="Date">Date</option>
          <option value="Amount">Amount</option>
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  };
};

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseListFilters);