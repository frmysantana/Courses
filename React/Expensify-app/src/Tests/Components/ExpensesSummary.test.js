import React from 'react';
import { ExpensesSummary } from '../../Components/ExpensesSummary';
import { shallow } from 'enzyme';
import expenses from '../Fixtures/expenses';
import { filters } from '../Fixtures/filters';

test('Should render one expense summary correctly', () => {
  const wrapper = shallow(<ExpensesSummary expensesLength={1} expensesTotal={1200}/>);
  expect(wrapper).toMatchSnapshot();
})

test('Should render with the count and total of multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesLength={14} expensesTotal={1600000} />);
  expect(wrapper).toMatchSnapshot();
});