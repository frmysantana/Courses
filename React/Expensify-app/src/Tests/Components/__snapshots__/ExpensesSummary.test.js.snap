// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render one expense summary correctly 1`] = `
<p>
  Viewing 
  1
   
  expense
   totalling 
  $12.00
  .
</p>
`;

exports[`Should render with the count and total of multiple expenses 1`] = `
<p>
  Viewing 
  14
   
  expenses
   totalling 
  $16,000.00
  .
</p>
`;
